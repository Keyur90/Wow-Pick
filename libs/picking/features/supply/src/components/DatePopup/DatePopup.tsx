import React from 'react';
import { format } from 'date-fns';
import { makeStyles, Typography, Dialog, DialogContent, DialogActions, Button } from '@material-ui/core';
import DateFnsUtils from '@date-io/date-fns';
import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';

interface DatePopupPropTypes {
  date: Date | string;
  isOpen: boolean;
  onOk: () => void;
  onClose: () => void;
  onChange: (event: Date) => void;
}

const useDialogContentStyles = makeStyles((theme) => ({
  root: {
    color: '#171C1F',
    '&:first-child': {
      padding: theme.spacing(2, 3, 0),
    },
    '& > h3': {
      fontSize: '1.25rem',
      lineHeight: '2rem',
      fontWeight: 500,
      marginBottom: theme.spacing(2),
    },
  },
}));
const useStyles = makeStyles((theme) => ({
  expiryDateField: {
    minWidth: '100%',
  },
}));
const useDialogActionStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(3),
  },
}));

const DatePopup: React.FC<DatePopupPropTypes> = ({ isOpen, onOk, onClose, onChange, date }) => {
  const dialogContentClasses = useDialogContentStyles();
  const dialogActionClasses = useDialogActionStyles();
  const classes = useStyles();

  return (
    <MuiPickersUtilsProvider utils={DateFnsUtils}>
      <Dialog
        onClose={(event, reason) => {
          if (reason !== 'backdropClick') onOk();
        }}
        open={isOpen}
        fullWidth
        maxWidth="sm"
      >
        <DialogContent classes={dialogContentClasses}>
          <Typography variant="h3">Expiry date</Typography>
          <KeyboardDatePicker
            className={classes.expiryDateField}
            disableToolbar
            format="dd/MM/yyyy"
            margin="normal"
            id="date-picker-dialog"
            label="Enter the expiry date"
            value={date}
            onChange={onChange}
            KeyboardButtonProps={{
              'aria-label': 'change date',
            }}
          />
        </DialogContent>
        <DialogActions classes={dialogActionClasses}>
          <Button color="secondary" variant="outlined" onClick={onClose}>
            CLOSE
          </Button>
          <Button variant="outlined" color="secondary" onClick={onOk}>
            OK
          </Button>
        </DialogActions>
      </Dialog>
    </MuiPickersUtilsProvider>
  );
};

export { DatePopup };
