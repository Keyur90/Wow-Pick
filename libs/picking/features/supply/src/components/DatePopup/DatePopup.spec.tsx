import React from 'react';
import { render } from '@testing-library/react';
import { DatePopup } from './DatePopup';

test('renders ExpiryDatePopup', () => {
  const { baseElement } = render(<DatePopup date={null} isOpen={true} onOk={null} onClose={null} onChange={null} />);

  expect(baseElement).toMatchSnapshot();
});
