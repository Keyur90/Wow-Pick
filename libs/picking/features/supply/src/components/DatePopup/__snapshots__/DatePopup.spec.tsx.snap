// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExpiryDatePopup 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper"
      role="none presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
        role="dialog"
      >
        <div
          class="MuiDialogContent-root makeStyles-root-1"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3"
          >
            Expiry date
          </h3>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-expiryDateField-3 MuiFormControl-marginNormal"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
              for="date-picker-dialog"
              id="date-picker-dialog-label"
            >
              Enter the expiry date
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                id="date-picker-dialog"
                type="text"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="change date"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            
          </div>
        </div>
        <div
          class="MuiDialogActions-root makeStyles-root-2 MuiDialogActions-spacing"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              CLOSE
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              OK
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
